docs/openapi-docs/src/test/resources/expected_external.yml

```diff
@@ -0,0 +1,17 @@
+openapi: 3.0.3
+info:
+  title: Fruits
+  version: '1.0'
+paths:
+  /:
+    get:
+      operationId: getRoot
+      responses:
+        '200':
+          description: ''
+        default:
+          description: ''
+          content:
+            application/json:
+              schema:
+                $ref: https://opensource.zalando.com/restful-api-guidelines/models/problem-1.0.1.yaml#/Problem
```

Comment:
value of `$ref` should be quoted like 'xxx'

Reply:
I agree. But this seems to be done somewhere in the YAML serializer. I have no idea how this works. Some help here would be appreciated. I'll have a look but I'm not confident that I can fix this.

Reply:
This depends on the String Style of the `TapirOpenAPICirceYaml.toYaml` method. In the `Printer` class in line 69 is this check:
```
  private def stringScalarStyle(value: String): DumperOptions.ScalarStyle =
    if (isBad(value)) DumperOptions.ScalarStyle.DOUBLE_QUOTED
    else if (stringStyle == StringStyle.Plain && hasNewline(value)) DumperOptions.ScalarStyle.LITERAL
    else StringStyle.toScalarStyle(stringStyle)
```
So I'm not sure how to fix this or if it's working as intended.

Reply:
Maybe the `Printer.isBad` method needs to be adjusted? I'm not sure...

Reply:
It seems the Printer needs to be modified somewhat: https://yaml.org/spec/1.2.2/#733-plain-style

Reply:
I think it's handled in such way in the underlying `snakeyaml` lib since it's used by circe `Printer`. I see that `Printer` is not changing the string style for url string. If you are convinced that this is incorrect handling as for yaml definition I think you can try to report issue in `snakeyaml`, I am not a yaml expert :D

Reply:
I'm also not an expert there. Do you think this PR can be merged then?

Reply:
Yes, thanks!